public/src/main/java/com/revenuecat/purchases/models/GooglePurchaseInfo.kt

```diff
@@ -0,0 +1,29 @@
+package com.revenuecat.purchases.models
+
+import com.android.billingclient.api.ProductDetails
+import com.revenuecat.purchases.ProductType
+
+sealed class GooglePurchaseInfo: PurchaseInfo {
+    data class NotSubscription(
+        override val productId: String,
+        val productDetails: ProductDetails
+    ) : GooglePurchaseInfo()
+
+    data class Subscription(
+        override val productId: String,
+        val productDetails: ProductDetails,
+
+        val optionId: String?,
+        val token: String
```

Comment:
Hmm I'm not a big fan that this info is duplicated in `GooglePurchaseOption`. But I've been thinking and couldn't come up with a better architecture, and since this is internal, I think we can go with this.